zephyr/scripts/clean-expected-release-differences.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/sh
+# SPDX-License-Identifier: BSD-3-Clause
+# shellcheck disable=SC3043
+
+set -e
+
+die()
+{
+    # shellcheck disable=SC2059
+    >&2 printf "$@"
+    exit 1
+}
+
+fix_dir()
+{
+    local bd="$1"
+
+    test -d "$bd"/build-sof-staging ||
+        die 'No %s/build-sof-staging directory\n' "$bd"
+
+    # config files have absolute paths
+    find "$bd" -name 'config.gz' -exec rm '{}' \;
+
+    # In case of a compression timestamp. Also gives better messages.
+    find "$bd" -name '*.gz' -print0 | xargs -r -0 gunzip
+
+    ( set -x
+
+      # Native binaries
+      rm -f "$bd"/build-sof-staging/tools/sof-logger*
+      # Python and other scripts
+      dos2unix "$bd"/build-sof-staging/tools/* || true
+
+      # signature salt
+      find "$bd" -name '*.ri' -exec rm '{}' \;
+
+      # debug symbols
+      find "$bd" -name main.mod   -exec rm '{}' \;
+      find "$bd" -name zephyr.elf -exec rm '{}' \;
+
+      # Fix absolute paths to SDK libraries,
+      # e.g.: /opt/toolchains/zephyr-sdk-0.15.2/xtensa-intel_s1000_zephyr-elf/bin/../lib/
+      #             gcc/xtensa-intel_s1000_zephyr-elf/12.1.0/libgcc.a(_divsf3.o)
+      #
+      # This helps A LOT so it's kept here for debugging but it's not
+      # enough: there are still some small differences left.
+      find "$bd" -name zephyr.map -exec sed -i'' -e \
+        's#[^[:blank:]]*zephyr-sdk-[^/]*/xtensa#ZSDK/xtensa#; s#\\#/#g; /^ \.debug_/ d' \
+        '{}' \;
```

Comment:
I'm a bit confused here.
Can you explain what it exactly does?

Reply:
The `zephyr.map` file has a lot of absolute paths. I'll rephrase the comments.